### PR TITLE
Update Notice and add cgmanifest.json

### PR DESCRIPTION
**Pull request by vhvb1989**
- Add component governance `cgmanifest.json`
- Update Notice

fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/23

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -88,7 +90,8 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 
-
+License notice for
```
Comment (vhvb1989) [inline NOTICE.txt:93]: @Jinming-Hu  @katmsft  can you mention what this license is from?

File: NOTICE.txt [hunk 2]
```diff
@@ -292,6 +295,8 @@ SOFTWARE.
    See the License for the specific language governing permissions and
    limitations under the License.
 
+License notice for
```
Comment (vhvb1989) [inline NOTICE.txt:298]: @Jinming-Hu  @katmsft  can you mention what this license is from?

File: NOTICE.txt [hunk 3]
```diff
@@ -315,30 +320,8 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
...
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-SOFTWARE.
-
+License notice for
```
Comment (vhvb1989) [inline NOTICE.txt:323]: @Jinming-Hu  @katmsft  can you mention what this license is from?

File: NOTICE.txt [hunk 4]
```diff
@@ -362,6 +345,8 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 
+License notice for
```
Comment (vhvb1989) [inline NOTICE.txt:348]: @Jinming-Hu  @katmsft  can you mention what this license is from?

File: NOTICE.txt [hunk 5]
```diff
@@ -385,6 +370,8 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
...
```
Comment (vhvb1989) [inline NOTICE.txt:373]: @Jinming-Hu  @katmsft  can you mention what this license is from?

File: NOTICE.txt [hunk 6]
```diff
@@ -408,6 +395,8 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 
+License notice for
```
Comment (vhvb1989) [inline NOTICE.txt:398]: @Jinming-Hu  @katmsft  can you mention what this license is from?

File: NOTICE.txt [hunk 7]
```diff
@@ -431,6 +420,8 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 
+License notice for amalgamate
```
Comment (vhvb1989) [inline NOTICE.txt:423]: @Jinming-Hu  @katmsft  is storage using Amalgamate?  for SQL lite?

**Review by vhvb1989 — COMMENTED** [review]
Requesting info about current licenses on Notice from Storage team

**Comment by vhvb1989** [comment]
Need to double-check with @RickWinter  if we will have Notice and cgmanifest per package and none on the repo root
https://github.com/Azure/azure-sdk-for-cpp/pull/687/files

**Comment by ahsonkhan** [comment]
> if we will have Notice and cgmanifest per package and none on the repo root

You certainly need it at the repo root. If we really want to, we _could_ optionally include a copy of the Notice.txt per package, but I don't think it is necessary at this time.

**Comment by vhvb1989** [comment]
@ahsonkhan 
> we could optionally include a copy of the Notice.txt per package

This would create a lot of sync issues. We would need to update the main Notice and then copy to all packages?  or update just one package and main Notice?  This is not ideal.

I think we should leave one Notice per package, and during Release, we can copy the Notice to the root (for that package) before creating the zip file.  Then we can remove from it everything else.

File: sdk/storage/azure-storage-blobs/NOTICE.txt [hunk 8]
```diff
@@ -30,41 +30,43 @@ bring it to our attention. Post an issue or email us:
...
-Copyright (c) 1996 - 2020, Daniel Stenberg, <daniel@haxx.se>
+BSD 3-Clause License
+
+Copyright (c) 2017, University of Cincinnati
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+* Redistributions of source code must retain the above copyright notice, this
```
Comment (antkmsft) [inline sdk/storage/azure-storage-blobs/NOTICE.txt:45]: IIRC, the copyright lines above ("Copyright (C) ...") should remain intact; and even maybe if we contribute, MS notice should go there too.

**Comment by ahsonkhan** [comment]
> This would create a lot of sync issues. We would need to update the main Notice and then copy to all packages? or update just one package and main Notice? This is not ideal.

To that point, in .NET, we created a **single copy at the root**, and every package being shipped to the package manager had a copy of that third party notice txt file embedded in it, as part of the publish step (include license txt):
image

We could follow a similar practice here. That said, since we are planning to publish sources directly from GitHub to Vcpkg, keeping just one at the root is fine.

**Comment by RickWinter** [comment]
> > This would create a lot of sync issues. We would need to update the main Notice and then copy to all packages? or update just one package and main Notice? This is not ideal.
> 
> To that point, in .NET, we created a **single copy at the root**, and every package being shipped to the package manager had a copy of that third party notice txt file embedded in it, as part of the publish step (include license txt):
> image
> 
> We could follow a similar practice here. That said, since we are planning to publish sources directly from GitHub to Vcpkg, keeping just one at the root is fine.

One at the root may not be accurate.  If KeyVault has a third-party dependency then we are shipping that when I get storage.  Although its a superset of what my dependencies are, its not accurate to what each package actually depends on.

File: sdk/storage/azure-storage-blobs/NOTICE.txt (hunk 8 above)
Comment (RickWinter) [reply sdk/storage/azure-storage-blobs/NOTICE.txt:45]: Correct, the NOTICE.txt needs to be an unedited and complete copy of the exact license.  That includes spelling errors must not be fixed if they exist in the original license.

**Comment by ahsonkhan** [comment]
> Although its a superset of what my dependencies are, its not accurate to what each package actually depends on.

I understand. Having a superset has been the norm in .NET, even for individually shipping packages. If this is a priority to address, and the trade-offs of doing so are worth it (around maintenance and keeping things in sync), go for it.

Starting off with the simplest approach that mimics what is generally expected on GitHub/OSS is also good :)

File: cmake-modules/DefinePlatform.cmake [hunk 9]
```diff
@@ -1,3 +1,6 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (ahsonkhan) [inline cmake-modules/DefinePlatform.cmake:1]: Any other files (outside this PR) that is missing this header?

File: sdk/core/performance-stress/cgmanifest.json [hunk 10]
```diff
@@ -0,0 +1,4 @@
+{
+  "Registrations": [
+  ]
+}
```
Comment (ahsonkhan) [inline sdk/core/performance-stress/cgmanifest.json:4]: When empty, I am not sure if we need the file at all.

File: sdk/storage/azure-storage-files-shares/LICENSE [hunk 11]
```diff
@@ -0,0 +1,21 @@
+    MIT License
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-shares/LICENSE:1]: Make these all .txt?
https://github.com/Azure/azure-sdk-for-cpp/issues/887

File: cmake-modules/AddGoogleTest.cmake [hunk 12]
```diff
@@ -3,6 +3,8 @@
 # Downloads GTest and provides a helper macro to add tests. Add make check, as well, which
 # gives output on failed tests without having to set an environment variable.
 #
+# Modified version from https://github.com/CLIUtils/cmake/commit/4e52e4d0bc2e9fd27171926d0b5d9f396dd8637c
+# License on Notice
```
Comment (ahsonkhan) [inline cmake-modules/AddGoogleTest.cmake:7]:
```suggestion
# License available in NOTICE.txt file.
```

**Review by antkmsft — APPROVED** [review]

**Review by RickWinter — APPROVED** [review]

**Comment by RickWinter** [comment]
@vhvb1989 Can you get this cleaned up and committed

**Pull request merged**